addons/web/static/src/views/fields/date/date_field.xml

```diff
@@ -9,6 +9,7 @@
             <DatePicker
                 t-props="props.pickerOptions"
                 date="date"
+                inputId="props.name"
```

Comment:
you should use props.id instead of props.name, because the same field could have several occurrences in the same view. The id is unique (it's basically the field name for the first occurrence, and the field name with a suffix for the n+1 occurrences). See CharField